catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -269,39 +275,36 @@ private String getRegistryId(Metacard mcard) {
                 .toString();
     }
 
-    public void init()
-            throws UnsupportedQueryException, SourceUnavailableException, FederationException,
-            PluginExecutionException {
+    public void init() {
+        try {
+            List<Metacard> registryMetacards;
+            Filter registryFilter = filterBuilder.attribute(Metacard.TAGS).is().like()
+                    .text(RegistryConstants.REGISTRY_TAG);
+            QueryImpl query = new QueryImpl(registryFilter);
+            query.setPageSize(1000);
+            QueryRequest request = new QueryRequestImpl(query);
 
-        List<Metacard> registryMetacards;
-        Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
-                .is()
-                .like()
-                .text(RegistryConstants.REGISTRY_TAG);
-        QueryImpl query = new QueryImpl(registryFilter);
-        query.setPageSize(1000);
-        QueryRequest request = new QueryRequestImpl(query);
+            request.getProperties().put(SecurityConstants.SECURITY_SUBJECT,
+                    Security.runAsAdmin(() -> Security.getInstance().getSystemSubject()));
 
-        request.getProperties()
-                .put(SecurityConstants.SECURITY_SUBJECT,
-                        Security.runAsAdmin(() -> Security.getInstance()
-                                .getSystemSubject()));
+            QueryResponse response = catalogFramework.query(request);
 
-        QueryResponse response = catalogFramework.query(request);
+            if (response == null) {
+                throw new PluginExecutionException(
```

Comment:
You'll never even log this fact. Probably should add a log to the catch block prior to the exec service schedule call.